monitoring/mongodb/dashboard.py

```diff
@@ -0,0 +1,1359 @@
+            expr=f'avg(mongodb_ss_connections{{job=~"{Jobs.shard_pattern}",conn_type=~"{conn_type}", rs_nm=~"$rs_nm"}})',
+            legendFormat=legend,
+        )
+        for conn_type, legend in [
+            ("current", "Current"),
+            ("available", "Available"),
+        ]
+    ]
+    + [
+        Target(
+            expr=f'avg(mongodb_ss_globalLock_activeClients_{client_type}{{job=~"{Jobs.shard_pattern}", rs_nm=~"$rs_nm"}})',
+            legendFormat=f"Active {client_type} clients",
+        )
+        for client_type in ["readers", "writers"]
+    ]
+    + [
+        Target(
+            expr=f'avg(mongodb_ss_transactions_current{txn_state}{{job=~"{Jobs.shard_pattern}",rs_nm=~"$rs_nm"}})',
+            legendFormat=f"{txn_state} transactions",
+        )
+        for txn_state in ["Active", "Inactive"]
+    ],
+    unit="none",
+    description="MongoDB Connections",
+)
+
+disk_reads = mongodb_timeseries(
+    "Disk Reads Completed",
+    [
+        Target(
+            expr=f'sum by(pod) (rate(container_fs_reads_total{{namespace="{NAMESPACE}", pod=~".*mongodb.*"}}[$__rate_interval]))',
+            legendFormat="{{pod}}",
+        )
+    ],
+    unit=UNITS.OPS_PER_SEC,
+)
+
+disk_writes = mongodb_timeseries(
+    "Disk Writes Completed",
+    [
+        Target(
+            expr=f'sum by(pod) (rate(container_fs_writes_total{{namespace="{NAMESPACE}", pod=~".*mongodb.*"}}[$__rate_interval]))',
+            legendFormat="{{pod}}",
+        )
+    ],
+    unit=UNITS.OPS_PER_SEC,
+)
+
+# Oplog Panels
+oplog_recovery = mongodb_timeseries(
+    "Oplog Recovery Window",
+    [
+        Target(
+            expr=f'mongodb_mongod_replset_oplog_head_timestamp{{job=~"{Jobs.shard_pattern}"}}-mongodb_mongod_replset_oplog_tail_timestamp{{job=~"{Jobs.shard_pattern}"}}',
+            legendFormat="{{ pod }}",
+        )
+    ],
+    unit=UNITS.SECONDS,
+    lineWidth=2,
+    fillOpacity=20,
+    showPoints="never",
+    description="Timespan window between newest and the oldest op in the Oplog collection.",
+    legendDisplayMode="table",
+    legendPlacement="right",
+    legendValues=["mean", "max", "min"],
+)
+
+avg_op_latency = mongodb_timeseries(
+    "Average Operation Latency",
+    [
+        Target(
+            expr=f'sum by(type) (\n  rate(mongodb_mongod_op_latencies_latency_total{{job=~"{Jobs.shard_pattern}"}}[$__rate_interval])\n) /\nsum by(type) (\n  rate(mongodb_mongod_op_latencies_ops_total{{job=~"{Jobs.shard_pattern}"}}[$__rate_interval])\n)',
+            legendFormat="{{type}}",
+        )
+    ],
+    unit="µs",
+    thresholds=THRESHOLDS["latency"],
+    lineInterpolation="smooth",
+    fillOpacity=20,
+    showPoints="auto",
+    description="Average latency of operations (classified by read, write, or (other) command)",
+)
+
+replication_lag = mongodb_timeseries(
+    "Replication Lag",
+    [
+        Target(
+            expr=MongoDBMetrics.replication_lag(),
+            legendFormat="{{pod}} <- {{name}}",
```

Comment:
`name` is very long, would be best to shorten it somehow.
also not very clear what the "arrow" represents : it may be better to simply report the lag of each (secondary) pod, which will thus be implicetely relative to the primary instance of that shard --> as label, just use the pod/job/name which tracks the (secondary) while lag is being measured